Still improving the UI

luquiceno opened the pull request:
For this improvement I have created a new component called **NavMenu** . This component has a button **Logout** to finish the session and a link **Home** to see the list of the screenshots.
Also I created a **CardScreenshot** component that allows to see all the screenshots that a user has, as a card. 
Each card has the image of the screenshot and a button to link the user to the source of the screenshot.

The following image shows how the main page of the application looks like so far.


<img width="730" alt="Screenshot_2" src="https://github.com/user-attachments/assets/49b4bb56-c565-45ad-af87-8718e667befd">

luquiceno committed:
Co-authored-by: sebasutp <sebasutp@users.noreply.github.com>

luquiceno committed:
UI continues to be improved.
I added the NavMenu component.
I created the CardScreenshot component to see the list of Screenshots
that a user has.
In this change the user can end the session by clicking on a button
in the NavMenu component.

luquiceno committed:
UI continues to be improved.

luquiceno committed:
UI continues to be improved.
Correcting the last commit to this file.

sebasutp commented on src/components/CardScreenshot.jsx:
```diff
@@ -0,0 +1,31 @@
+function Card({ urlImage, image }) {
+  return (
+    <div>
+      <div className="w-72 h-72 bg-white border-2 border-gray-200 rounded-lg shadow dark:bg-gray-800 dark:border-gray-700 mb-8">
```
Is it possible to move all the style into a separate style file? It is hard to read the code otherwise

luquiceno replied:
I will find out if it can be moved, or what can be done about it. If not I will organize the code better to make it more readable.

sebasutp commented on src/components/Header.jsx:
```diff
@@ -0,0 +1,28 @@
+
+function Header() {
+
+  const [loggeado, setLoggeado] = useState(false)
```
Let's use a single language for consistency. Do not mix Spanish and English for variable names. Let's keep all of them in English

luquiceno replied:
I forgot to delete this component. I have replaced it for the NavMenu.

sebasutp commented on src/components/PluginScreenshot.jsx:
```diff
@@ -37,16 +35,29 @@ function NewScreenshotPlugin() {
-        id="urlInput"/>
-      <div>
-        <img id="screenshotImg" width="50%" />
+    <div className="flex flex-col items-center justify-center">
```
Since you also changed the PluginScreenshot view, can you add a screenshot of how it looks like now?

luquiceno replied:
I created a commit with the new changes and will add the image to show how it looked like.

sebasutp commented on src/components/NewScreenshot.jsx:
```diff
@@ -5,7 +5,7 @@ function NewScreenshot() {
   const [image, setImage] = useState(null);
 
-  const handleSubmit = async (e) => {
+ const handleSubmit = async (e) => {
```
Why removing the whitespace here? Let's keep the indentation as it should be :)

luquiceno replied:
I will consider the indentation for other commits. Maybe I moved it by accident.

sebasutp commented on src/components/NavMenu.jsx:
```diff
@@ -0,0 +1,59 @@
+            }
+                     
+            <button data-collapse-toggle="navbar-sticky" type="button" className="inline-flex items-center p-2 w-10 h-10 justify-center text-sm text-gray-500 rounded-lg md:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600" aria-controls="navbar-sticky" aria-expanded="false">
+              <span className="sr-only">Open main menu</span>
```
What does this button do? Could you add some screenshots of the nav-bar before and after log-in?

luquiceno replied:
The Navbar only appears when a user logs in. 

<img width="593" alt="Screenshot_3" src="https://github.com/user-attachments/assets/5dccccb5-282f-4b97-a6b6-3e047a44f7ff">

The ` <button data-collapse-toggle="navbar-sticky"`, let the user on small screens to see the linking options available in the Navbar. In this case and until now, we  have only **Home**.

The button marked with the red arrow is used in responsive applications. 
The Navbar looks like this on small screens. 


<img width="462" alt="Screenshot_5" src="https://github.com/user-attachments/assets/8f26cc85-90f5-4769-9f95-74d5fdf28a2c">

This tag `<span className="sr-only">Open main menu</span>` is for  screen reader users.

sebasutp commented on src/components/Login.jsx:
```diff
@@ -50,7 +50,7 @@ function Login() {
               <form 
                 className="space-y-4 md:space-y-6" 
                 onSubmit={handleSubmit}
-                action="#">
```
I think this action="#" might be necessary. Otherwise this submit button: https://github.com/sebasutp/screenshot_frontend/blob/89a6f46b67ac36457aeb1266fe17a57d1f2672ef/src/components/Login.jsx#L95-L98

Might create a default "post" action.

luquiceno replied:
This property `action="#"` on the form is not required, as it was not needed before.

sebasutp committed:
Merge branch 'main' into feature

sebasutp reviewed (changes requested):
If you could move all the style part into a separate file it would improve a lot the readability of the code. Otherwise it is hard to understand the changes to the UI and the logic separately. If it is too hard then just leave it as is.

luquiceno committed:
Improving the Screenshots UI
Improved user interface where you can see the screenshot taken and the
link from where it was taken.
Also the user interface where the screenshot just taken is displayed
and can be saved in the list of screenshots.
I deleted the Header component, since I had already changed it for the
NavMenu.
I indented the code to make it more readable.

luquiceno committed:
Merge branch 'feature' of https://github.com/luquiceno/screenshot_frontend_seb into feature

sebasutp commented on src/components/Screenshots.jsx:
```diff
@@ -1,16 +1,18 @@
     setIsLoading(true);
     const url = `${import.meta.env.VITE_BACKEND_URL}/screenshots/`;
-    console.log(token)  
+    console.log(token) 
```
Remove.

sebasutp commented on src/components/PluginScreenshot.jsx:
```diff
@@ -10,6 +10,7 @@ function NewScreenshotPlugin() {
       url: document.getElementById('urlInput').value,
       img: document.getElementById('screenshotImg').src
     };
+    console.log("esta es la url",url)
```
remove

sebasutp commented on src/components/NewScreenshot.jsx:
```diff
@@ -1,13 +1,13 @@
+  const handleSubmit = async (e) => {
     e.preventDefault();
-
+    //if (!image) return;
```
remove

sebasutp commented on src/components/Screenshots.jsx:
```diff
@@ -27,31 +29,44 @@ function Screenshots() {
+          <div>
+            <NavMenu />
+            <div className="flex flex-col items-center justify-center mt-24">
+              <p className="text-xl font-semibold text-gray-700 mb-16">
```
If we are already showing the screenshots, maybe we do not need to put again how many we have. Consider removing?

sebasutp commented on src/index.css:
```diff
@@ -1,3 +1,73 @@
\ No newline at end of file
+@tailwind utilities;
+
+/* :root {
```
Remove all the commented out stuff

sebasutp commented on src/main.jsx:
```diff
@@ -1,3 +1,4 @@
+//import { StrictMode } from 'react'
```
Remove

sebasutp commented on src/App.css:
```diff
@@ -0,0 +1,43 @@
+/* #root {
```
Remove? Maybe we can remove the entire file if we are not using it

sebasutp reviewed (commented):
The code looks a lot better. Please address the new comments and I will approve

luquiceno committed:
Addressing reviews.

luquiceno commented:
This is the view from the screenshot in the route "/item/{id}"

<img width="551" alt="Screenshot_6" src="https://github.com/user-attachments/assets/c335ec3b-f74f-475f-995a-7104cb6ef9e3">

And this is the view for uploading a screenshot using the plugin.

<img width="438" alt="Screenshot_7" src="https://github.com/user-attachments/assets/7cc1257a-4206-4fcf-9f42-6466df71af26">

luquiceno committed:
small bug fix.
